src/edu/csus/ecs/pc2/util/ScoreboardVariableReplacer.java

```diff
@@ -0,0 +1,131 @@
+// Copyright (C) 1989-2021 PC2 Development Team: John Clevenger, Douglas Lane, Samir Ashoo, and Troy Boudreau.
+package edu.csus.ecs.pc2.util;
+
+import edu.csus.ecs.pc2.core.CommandVariableReplacer;
+import edu.csus.ecs.pc2.core.model.Account;
+import edu.csus.ecs.pc2.core.model.ClientId;
+import edu.csus.ecs.pc2.core.model.Group;
+import edu.csus.ecs.pc2.core.model.IInternalContest;
+
+/**
+ * Substitute scoreboard team values for variables.
+ * 
+ * @author Douglas A. Lane <pc2@ecs.csus.edu>
+ *
+ */
+public class ScoreboardVariableReplacer {
+
+    public static final String COUNTRY_CODE = "{:countrycode}";
+
+    public static final String LONG_SCHOOL_NAME = "{:longschoolname}";
+
+    public static final String SHORT_SCHOOL_NAME = "{:shortschoolname}";
+
+    public static final String GROUP_NAME = "{:groupname}";
+
+    public static final String GROUP_ID = "{:groupid}";
+
+    public static final String EXTERNAL_ID = "{:externalid}";
+
+    public static final String CLIENT_NUMBER = "{:clientnumber}";
+
+    public static final String TEAM_LOGIN_NAME = "{:teamloginname}";
+
+    public static final String SITE_NUMBER = "{:sitenumber}";
+
+    public static final String TEAM_NAME = "{:teamname}";
+
+    public static final String[] VARIABLE_NAMES = { //
+            TEAM_NAME, //
+            SITE_NUMBER, //
+            TEAM_LOGIN_NAME, //
+            CLIENT_NUMBER, //
+            EXTERNAL_ID, //
+            GROUP_ID, //
+            GROUP_NAME, //
+            SHORT_SCHOOL_NAME, //
+            LONG_SCHOOL_NAME, //
+            COUNTRY_CODE, //
+    };
+
+    /**
+     * Substitutes values for variables.
+     * 
+     * 
+     * Variable names and examples.
+     * 
+     * <pre>
+     * Client/Team number - {:clientnumber} = 514
+     * Country Code - {:countrycode} = CAN
+     * CMS/External ID - {:externalid} = 309407
+     * Group Id Number - {:groupid} = 12545
+     * Group name - {:groupname} = Canada - University of British Columbia D1
+     * Long School name - {:longschoolname} = UBC! (U British Columbia)
```

Comment:
This does not seem to me to be a good example of a `Long School name`.  This is an example of a _combination_ of _two_ things:  a `Team Name` ("UBC!") and a `Short School name` ("U British Columbia").  An example of a `Long School name` would be something like `University of British Columbia` or `California State University, Sacramento`  (whose corresponding `Short School names` would be `U British Columbia` and `CSUS`, respectively).

Reply:
no change.

Reply:
I don't understand why you think a combination of a team name and a short school name (which is clearly what this is, based on the OTHER examples) is a good example of a Long School Name?

Reply:
may be change later.